include/SFML/Window/WindowBase.hpp

```diff
@@ -220,11 +220,56 @@ class SFML_WINDOW_API WindowBase
     ///
     /// \return The event, otherwise `std::nullopt` on timeout or if window was closed
     ///
-    /// \see pollEvent
+    /// \see pollEvent, handleEvents
     ///
     ////////////////////////////////////////////////////////////
     [[nodiscard]] std::optional<Event> waitEvent(Time timeout = Time::Zero);
 
+    ////////////////////////////////////////////////////////////
+    /// \brief Handle all pending events
+    ///
+    /// This function is not blocking: if there's no pending event then
+    /// it will return without calling any of the handlers.
+    ///
+    /// This function takes a variadic list of event handlers that
+    /// each take a concrete event type as a single parameter. The
+    /// event handlers can be any kind of callable object that has
+    /// an operator() defined for a specific event type.
+    ///
+    /// Examples:
+    /// - Lambda expressions: `[&](const sf::Event::KeyPressed) { ... }`
+    /// - Free functions: `void handler(const sf::Event::KeyPressed&) { ... }`
+    ///
+    /// Calling member functions is supported through lambda
+    /// expressions.
```

Comment:
I know what it's meant to express, but at the same time it feels kind of (un-)specific, too. Shouldn't it work with something like `std::mem_fn()`, too?

Reply:
`mem_fn` doesn't really work here. The provided callable has to be callable when passed an event object and currently the event object doesn't have any member functions that would be useful in such a context. In general, lambda expressions can be used everywhere where `mem_fn` would have been used anyway.